applications/luci-app-openwisp/htdocs/luci-static/resources/view/openwisp.js

```diff
@@ -0,0 +1,101 @@
+        o.rmempty = false
+
+        o = s.taboption('advanced', widgets.DeviceSelect, 'mac_interface', _('MAC Interface'), _('The interface to use for getting the MAC for this device.'))
+        o.rmempty = false
+
+        o = s.taboption('advanced', widgets.NetworkSelect, 'mac_interface', _('Management Interface'), _('The interface to use for management and control.'))
+        o.rmempty = false
+
+        o = s.taboption('advanced', form.Flag, 'merge_config', _('Merge Config'), _('If selected, in the event a config item is present in both the remote and local configuration, the remote configuration takes prescedence over local configuration.'))
+        o.rmempty = false
+
+        o = s.taboption('advanced', form.DynamicList, 'tags', _('Tags'), _('Tags applied to this device.'))
+
+        o = s.taboption('advanced', form.Flag, 'test_config', _('Test Config'), _('If the agent is unable to reach the controller after applying the downloaded config it will be reverted.'))
+
+        o = s.taboption('advanced', form.Value, 'test_script', _('Test Script'), _('Path to a custom test script if the default Test Config script does not meet your needs.'))
+        o.depends({ test_config: '1' })
+        o.datatype = "file"
+
+        o = s.taboption('advanced', form.Flag, 'hardware_id_key', _('Hardware ID Key'), _('Use a unique hardware ID for device identification, for example a serial number.'))
+        o.rmempty = false
+
+        o = s.taboption('advanced', form.Value, 'hardware_id_script', _('Hardware ID Script'), _('Path to the script used to return the value of the hardware key.'))
+        o.depends({ hardware_id_key: '1' })
+        o.datatype = "file"
+
+        o = s.taboption('advanced', form.Value, 'bootup_delay', _('Bootup Delay'), _('Maximum value of the bootup delay before starting OpenWISP service. Expressed in seconds.'))
+        o.placeholder = '0'
+        o.datatype = 'uinteger';
+
+        o = s.taboption('advanced', form.Value, 'connect_timeout', _('Connect Timeout'), _('Value passed to curl --connect-timeout argument, defaults to 15. Expressed in seconds.'))
+        o.placeholder = '15'
+        o.datatype = 'uinteger';
+
+        o = s.taboption('advanced', form.Value, 'max_time', _('Max Time'), _('Value passed to curl --max-time argument, defaults to 30. Expressed in seconds.'))
+        o.placeholder = '30'
+        o.datatype = 'uinteger';
+
+        o = s.taboption('advanced', form.Value, 'capath', _('CA Path'), _('Path passed to curl --capath argument, by default is empty.'))
+        o.datatype = "file"
+
+        o = s.taboption('advanced', form.Value, 'capath', _('CA Cert'), _('Path passed to curl --cacert argument, by default is empty.'))
+        o.datatype = "file"
```

Comment:
Do we need to mention in the `description` that we will use these options in Curl?
for example `Path passed to curl --cacert argument, by default is empty.`
That is not relevant isn`t it?

Reply:
You're right, probably an irrelevant detail.  I have changed it to what is in the curl documentation, since that is more to the point.